src/Template/Loggers/IOHprofiler_ecdf_logger.hpp

```diff
@@ -191,12 +191,44 @@ const IOHprofiler_AttainSuite& IOHprofiler_ecdf_logger<T>::data()
 }
 
 template<class T>
-const IOHprofiler_AttainMat& IOHprofiler_ecdf_logger<T>::at(size_t problem_id, size_t instance_id, size_t dim_id) const
+const IOHprofiler_AttainMat& IOHprofiler_ecdf_logger<T>::at(size_t problem_id, size_t instance_id, size_t dim_id, size_t run_id) const
 {
     assert(_ecdf_suite.count(problem_id) != 0);
     assert(_ecdf_suite.at(problem_id).count(dim_id) != 0);
     assert(_ecdf_suite.at(problem_id).at(dim_id).count(instance_id) != 0);
-    return _ecdf_suite.at(problem_id).at(dim_id).at(instance_id);
+    assert(_ecdf_suite.at(problem_id).at(dim_id).at(instance_id).count(run_id) != 0);
+    return _ecdf_suite.at(problem_id).at(dim_id).at(instance_id).at(run_id);
+}
+
+template<class T>
+IOHprofiler_AttainMat IOHprofiler_ecdf_logger<T>::at(size_t problem_id, size_t instance_id, size_t dim_id) const
+{
+    size_t runs;
+    IOHprofiler_AttainMat mat = _empty;
+    std::vector< std::vector< int >> c_mat(_range_error.size(),std::vector<int>( _range_evals.size(),0));
+    size_t ibound = _range_error.size();
+    size_t jbound = _range_evals.size();
+
+    assert(_ecdf_suite.count(problem_id) != 0);
+    assert(_ecdf_suite.at(problem_id).count(dim_id) != 0);
+    assert(_ecdf_suite.at(problem_id).at(dim_id).count(instance_id) != 0);
+
+    runs = _ecdf_suite.at(problem_id).at(dim_id).at(instance_id).size();
+    for(size_t i = 0; i < ibound; ++i) {
+        for(size_t j = 0; j < jbound; ++j) {
+            for(size_t r = 0; r < runs; ++r) {
+                c_mat[i][j] += _ecdf_suite.at(problem_id).at(dim_id).at(instance_id).at(r)[i][j]==true?1:0;
+            }  // for r
+        } // for j
+    } // for i
+    
+    for(size_t i = 0; i < ibound; ++i) {
+        for(size_t j = 0; j < jbound; ++j) {
+            mat[i][j] = (double)c_mat[i][j] / (double)runs > 0.5;
```

Comment:
It would be clearer and less error-prone to use `static_cast<double>(…)` instead of C-style cast.

Also, the `0.5` constant should not be hard-coded here, but exposed as a parameter within the constructor.